src/Appwrite/Platform/Workers/StatsUsage.php

```diff
@@ -2,24 +2,99 @@
 
 namespace Appwrite\Platform\Workers;
 
-use Appwrite\Event\StatsUsageDump;
 use Exception;
+use Throwable;
 use Utopia\CLI\Console;
 use Utopia\Database\DateTime;
 use Utopia\Database\Document;
 use Utopia\Platform\Action;
 use Utopia\Queue\Message;
+use Utopia\Registry\Registry;
 use Utopia\System\System;
 
 class StatsUsage extends Action
 {
+    /**
+     * In memory per project metrics calculation
+     */
     private array $stats = [];
     private int $lastTriggeredTime = 0;
     private int $keys = 0;
     private const INFINITY_PERIOD = '_inf_';
     private const BATCH_SIZE_DEVELOPMENT = 1;
     private const BATCH_SIZE_PRODUCTION = 10_000;
 
+    /**
+    * Stats for batch write separated per project
+    * @var array
+    */
+    private array $projects = [];
+
+    /**
+     * Array of stat documents to batch write to logsDB
+     * @var array
+     */
+    private array $statDocuments = [];
+
+    protected Registry $register;
+
+    /**
+     * Metrics to skip writing to logsDB
+     * As these metrics are calculated separately
+     * by logs DB
+     * @var array
+     */
+    protected array $skipBaseMetrics = [
+        METRIC_DATABASES => true,
+        METRIC_BUCKETS => true,
+        METRIC_USERS => true,
+        METRIC_FUNCTIONS => true,
+        METRIC_TEAMS => true,
+        METRIC_MESSAGES => true,
+        METRIC_MAU => true,
+        METRIC_WEBHOOKS => true,
+        METRIC_PLATFORMS => true,
+        METRIC_PROVIDERS => true,
+        METRIC_TOPICS => true,
+        METRIC_KEYS => true,
+        METRIC_FILES => true,
+        METRIC_FILES_STORAGE => true,
+        METRIC_DEPLOYMENTS_STORAGE => true,
+        METRIC_BUILDS_STORAGE => true,
+        METRIC_DEPLOYMENTS => true,
+        METRIC_BUILDS => true,
+        METRIC_COLLECTIONS => true,
+        METRIC_DOCUMENTS => true,
+        METRIC_DATABASES_STORAGE => true,
+    ];
+
+    /**
+     * Skip metrics associated with parent IDs
+     * these need to be checked individually with `str_ends_with`
+     */
+    protected array $skipParentIdMetrics = [
+        '.files',
+        '.files.storage',
+        '.collections',
+        '.documents',
+        '.deployments',
+        '.deployments.storage',
+        '.builds',
+        '.builds.storage',
+        '.databases.storage'
+    ];
+
+    /**
+     * @var callable
```

Comment:
You can define this like:

```
    * @var callable(): Database
```

Same for function params, and then you don't need to add `/** @var Database $dbForLogs*/` on the usage, because the IDE can already infer it's a `Database`